server/repository/src/db_diesel/stock_line.rs

```diff
@@ -214,6 +221,21 @@ fn create_filtered_query(filter: Option<StockLineFilter>) -> BoxedStockLineQuery
                 .select(location_dsl::id.nullable());
             query = query.filter(stock_line_dsl::location_id.eq_any(location_ids));
         }
+
+        if master_list.is_some() {
+            let mut master_list_filter = master_list.unwrap_or_default();
+            master_list_filter = master_list_filter
+                .exists_for_store_id(EqualFilter::equal_to(store.as_deref().unwrap_or_default()));
+
+            let master_list_ids =
+                MasterListRepository::create_filtered_query(Some(master_list_filter))
+                    .select(master_list_dsl::id);
+            let item_link_ids = master_list_line_dsl::master_list_line
```

Comment:
Would prefer if master list line also accepted master list filter, I was actually suggesting that this should be used in the pricing work, for examples I think this can be simplified : https://github.com/msupply-foundation/open-msupply/blob/db7e48c26b37fa214f7a69b55f9b8b0fdc2ff123/server/service/src/pricing/item_price.rs#L27-L51

to just:

```rust
    let default_price_per_unit = MasterListLineRepository::new(&ctx.connection)
        .query_by_filter(
            MasterListLineFilter::new()
                .master_list(MasterListFilter::new().is_default_price_list(true))
                .item_id(EqualFilter::equal_to(&input.item_id)),
        )?
        .pop()
        .map(|l| l.price_per_unit);
```